### PR TITLE
Add formatting to automatically clean notebooks

### PR DESCRIPTION
**Pull request by charlesbvll**
#### What does this implement/fix? Explain your changes.

These formatting scripts would avoid people from committing notebooks containing metadata that can potentially harm the version control process.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -94,6 +94,7 @@ sphinxcontrib-mermaid = "==0.7.1"
 furo = "==2022.12.7"
 sphinx-reredirects = "==0.1.1"
 nbsphinx = "==0.8.12"
+nbstripout = "^0.6.1"
```
Comment (danieljanes) [inline pyproject.toml:97]:
```suggestion
nbstripout = "==0.6.1"
```

File: dev/format.sh [hunk 2]
```diff
@@ -14,3 +14,9 @@ find src/proto/flwr/proto -name *.proto | grep "\.proto" | xargs clang-format -i
 # Examples
 python -m black -q examples
 python -m docformatter -i -r examples
+
+
+# Notebooks
+KEYS="metadata.celltoolbar metadata.language_info metadata.toc metadata.notify_time metadata.varInspector metadata.accelerator metadata.vscode cell.metadata.id cell.metadata.heading_collapsed cell.metadata.hidden cell.metadata.code_folding cell.metadata.tags cell.metadata.init_cell"
+python -m nbstripout doc/source/tutorial/*.ipynb --extra-keys "$KEYS"
+python -m nbstripout examples/*/*.ipynb --extra-keys "$KEYS"
```
Comment (danieljanes) [inline dev/format.sh:22]: Can we add these in "check-only mode" to `test.sh`?

**Review by danieljanes — CHANGES_REQUESTED** [review]
Lgtm, we can almost merge it

File: dev/format.sh (hunk 2 above)
Comment (charlesbvll) [reply dev/format.sh:22]: @danieljanes Well I looked into it, there is an option to just check if the files would need formatting but I don't think there is an easy way to halt the execution of the script if so.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**